R/interventions.R

```diff
@@ -0,0 +1,154 @@
+default_intervention_params <- list(
+  #######################
+  ## LLINs
+  ## At each timepoint, x% of the population receive a new net.
+  ## If they have an existing net it is replaced.
+  LLIN_years = c(), #eg, c(2002, 2005, 2008, 2011, 2015, 2018, 2021, 2024, 2027, 2030, 2033, 2036, 2039, 2042)
+  LLIN_cover = c(), #eg, c(0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.8, 0.8)
+
+  #######################
+  ## IRS
+  ## At each timepoint, x% of the population is protected by spraying
+  ## Note the model does not have household structure and protection
+  ## status is on the individual level
+  IRS_years = c(), #eg, c(2002, 2010)
+  IRS_cover = c(), #eg, c(0.5, 0.5)
+
+  ###############################
+  ## First-line treatment of clinical episodes with 
+  ## a blood-stage drug.
+  ## Includes information for coverage, efficacy and
+  ## duration of prophylaxis
+  BS_treat_year_on  =  c(), #eg  c(2000 ,2010)
+  BS_treat_year_off =  c(), #eg c(2010, 2018.5)
+  BS_treat_cover    =  c(), #eg c(0.5, 0.8)
+  BS_treat_BSeff    =  c(), #eg c(0.95, 0.95)
+  BS_treat_BSproph  =  c(), #eg c(10, 28)
+
+  ###############################
+  ## First-line treatment of clinical episodes with 
+  ## a hypnozoiticidal drug (tafenoquine or primaquine).
+  ## Includes information for coverage, efficacy and
+  ## duration of prophylaxis
+  PQ_treat_year_on  = c(), #eg c( 2010, 2021 )
+  PQ_treat_year_off = c(), #eg c( 2020, 2025 )
+  PQ_treat_cover    = c(), #eg c( 0.9, 0.8 )
+  PQ_treat_PQcover  = c(), #eg c( 1, 0.9 )
+  PQ_treat_BSeff    = c(), #eg c( 0.95, 0.7 )
+  PQ_treat_PQeff    = c(), #eg c( 0.95, 0.8 )
+  PQ_treat_BSproph  = c(), #eg c( 14, 60 )
+  PQ_treat_PQproph  = c(), #eg c( 14, 60 ) 
+  PQ_treat_CYP2D6   = c()  #eg c( 1, 0 )
+)
+
+#'@title Generate the intervention file
+#'@description to write a file to parameterise the interventions in the model
+#'
+#'@param overrides, the intervention model params that take precedence over the
+#'default parameters
+#'@param outpath, the path to write the intervention parameters to
+generate_intervention_file <- function(overrides, outpath) {
+  params <- default_intervention_params
+  for (name in names(overrides)) {
+    params[[name]] <- overrides[[name]]
+  }
+
+  if(length(params$LLIN_years) != length(params$LLIN_cover)) {
+    stop('LLIN vectors are not the same size')
+  }
+
+  if(length(params$IRS_years) != length(params$IRS_cover)) {
+    stop('IRS vectors are not the same size')
+  }
+
+  years <- sort(unique(c(
+    params$LLIN_years,
+    params$IRS_years,
+    params$BS_treat_year_on,
+    params$PQ_treat_year_on
+  )))
+
+  if (length(years) == 0) {
+    INT_cov <- matrix(-1, nrow=1, ncol=24)

```

Comment:
Capitilised INT_cov is easy to mis-read as ITN (bednet) coverage. Perhaps it can be renamed for clarity?